.github/workflows/inspektor-gadget.yml

```diff
@@ -17,12 +17,12 @@ jobs:
     # level: 0
     runs-on: ubuntu-latest
     steps:
-    - name: Setup go 1.17
+    - name: Setup Go
       uses: actions/setup-go@v1
       with:
-        go-version: 1.17
+        go-version: 1.18
       id: go
-    - name: Cache go 1.17
+    - name: Cache Go
```

Comment:
The new version of `actions/setup-go` can manage the Go cache automatically, so we don't need this section anymore. 
This requires to move the `checkout` above the `setup-go`:
https://github.com/alban/inspektor-gadget/commit/2126f17d80f77eeb1628e844eee592fa3b932191

Reply:
It is being managed in https://github.com/kinvolk/inspektor-gadget/pull/696.